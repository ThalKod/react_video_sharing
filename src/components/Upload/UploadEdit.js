import React from "react";
import { WithContext as ReactTags } from 'react-tag-input';

import { request, formatBytes, formatSecond } from "../../utils";
import LoadingSpinner from "../LaodingSpinner";

class UploadEdit extends React.Component{

  state = {
    name: "",
    size: 0,
    duration: 0,
    coverPhoto: null,
    loading: true,
    description: "",
    tags: [],
  };

  handleTagDelete = (i) => {
    const { tags } = this.state;
    this.setState({
      tags: tags.filter((tag, index) => index !== i),
    });
  };

  handleTagAddition = (tag) => {
    this.setState(state => ({ tags: [...state.tags, tag] }));
  };

 handleTagDrag = (tag, currPos, newPos) => {
   const { tags } = this.state;
   const tagsArr = [...tags];
   const newTags = tagsArr.slice();

   newTags.splice(currPos, 1);
   newTags.splice(newPos, 0, tag);

   // re-render
   this.setState({ tags: newTags });
 };

  componentDidMount = () => {
    const { videoId } = this.props;
    const getBasicInfo = request("get", `/video/basic/${videoId}`);
    const getImage = request("get", `/video/cover/default/${videoId}`);

    Promise.all([getBasicInfo, getImage])
        .then(res => this.setState({ ...res[0].data.video, ...res[1].data, loading: false }))
        .catch(err => console.log(err));
  };

  render(){
    const {
      name,
      size,
      duration,
      coverPhoto,
      loading,
      description,
      tags} = this.state;

    if(loading) return <LoadingSpinner/>;

    return (
         <div className="content-wrapper upload-page edit-page">
           <div className="container-fluid u-details-wrap">
             <div className="row">
               <div className="container">
                 <div className="col-lg-12">
                   <div className="u-details">
                     <div className="row">
                       <div className="col-lg-12 ud-caption">Upload Details</div>
                       <div className="col-lg-2">
                         <div className="imgplace">
                           <img src={coverPhoto} alt="cover" />
                         </div>
                       </div>
                       <div className="col-lg-10">
                         <div className="u-title">{name}</div>
                         <div className="u-size"><b>Size :</b> {formatBytes(size)} <b>Length:</b> {formatSecond(duration)}</div>
                         <div className="u-progress">
                           <div className="progress">
                             <div className="progress-bar" role="progressbar" aria-valuenow="100" aria-valuemin="0"
                                  aria-valuemax="100" style={{"width": "100%"}}>
                               <span className="sr-only">100% Complete</span>
                             </div>
                           </div>
                           <div className="u-close">
                             <a href="/"><i className="cvicon-cv-cancel" /></a>
                           </div>
                         </div>
                         <div className="u-desc">Your video is uploaded, please edit, add tags and save.</div>
                       </div>
                     </div>
                   </div>
                 </div>
               </div>
             </div>
           </div>
           <div className="container">
             <div className="row">
               <div className="col-lg-12">
                 <div className="u-form">
                   <div className="row">
                     <div className="col-lg-12">
                       <div className="form-group">
                         <label htmlFor="e1">Video Title</label>
                         <input
                             type="text"
                             className="form-control"
                             id="e1"
                             onChange={(e) => this.setState({ name: e.target.value })}
                             value={name}
                         />
                       </div>
                     </div>
                     <div className="col-lg-12">
                       <div className="form-group">
                         <label htmlFor="e2">About</label>
                         <textarea
                             onChange={e => this.setState({ description: e.target.value})}
                             value={description}
                             placeholder="Description of your video"
                             className="form-control"
                             name="e2"
                             id="e2"
                             rows="3"
                         />
                       </div>
                     </div>
                     <div className="col-lg-12">
                       <div className="form-group">
                         <ReactTags
                             tags={tags}
                             className="form-control"
                             handleDelete={this.handleTagDelete}
                             handleAddition={this.handleTagAddition}
                             handleDrag={this.handleTagDrag}
                             delimiters={[188, 13]}
                         />
                       </div>
                     </div>
                   </div>
                 </div>
                 <div className="u-area">
                     <button className="btn btn-primary u-btn" type="submit">Save</button>
                 </div>
                 <div className="u-terms">
                   <p>By submitting your videos to us, you acknowledge that you agree to circle's <a href="/">Terms
                     of Service</a> and <a href="/">Community Guidelines</a>.</p>
                   <p>Please be sure not to violate others' copyright or privacy rights. Learn more</p>
                 </div>
               </div>
             </div>
           </div>
         </div>
    )
  }
}


export default UploadEdit;
