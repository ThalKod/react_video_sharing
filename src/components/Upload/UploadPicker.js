import React from "react";
import ReactFilestack from 'filestack-react';


import requireLoggedIn from "../HOC/requireLoggedIn";

const UploadPicker = () => {

    const options = {
      "maxFiles": 1,
      "accept": [
        "video/mp4",
        "video/x-m4v",
        "video/*"
      ],
      "storeTo": {
        "container": "devportal-customers-assets",
        "path": "user-uploads/",
        "region": "us-east-1"
      },
      "fromSources": [
        "local_file_system",
        "url",
        "googledrive",
        "dropbox"
      ],
    };

    return(
        <div className="content-wrapper">
            <div className="container">
              <div className="row">
                <div className="col-lg-12 upload-page">
                  <div className="u-area">
                    <ReactFilestack
                      apikey={process.env.REACT_APP_API_KEY}
                      options={options}
                      onSuccess={(res) => { console.log("upload Res", res)}}
                      onError={(err) => {console.log("Upload err", err)}}
                      render={({ onPick }) => (
                        <div>
                          <i className="cv cv cvicon-cv-upload-video"/>
                          <p className="u-text1">Select Video files to upload</p>
                          <button type="submit" onClick={onPick} className="btn btn-primary u-btn">Upload Video</button>
                        </div>
                      )}
                    />
                  </div>

                  <div className="u-terms">
                    <p>By submitting your videos to circle, you acknowledge that you agree to circle's <a href="/">Terms
                      of Service</a> and <a href="/">Community Guidelines</a>.</p>
                    <p>Please be sure not to violate others' copyright or privacy rights. Learn more</p>
                  </div>
                </div>
              </div>
            </div>
        </div>
    )
};

export default requireLoggedIn(UploadPicker);


