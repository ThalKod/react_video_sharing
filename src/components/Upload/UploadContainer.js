import React from "react";

import UploadEdit from "./UploadEdit"
import UploadPicker from "./UploadPicker";
import requireLoggedIn from "../HOC/requireLoggedIn";
import LoadingSpinner from "../LoadingSpinner";
import { request } from "../../utils";


export class UploadContainer extends React.Component{
  state = {
    uploaded: false,
    loading: false,
    videoId: null
  };

  onUploadSuccess = (video) =>{
    this.setState({ uploaded: true, loading: true});
    request("post", "/upload/video", {}, video)
        .then(res => {
          console.log("video resolve+++",res);
          if(!res.error) return this.setState({ videoId: res.data.id, loading: false});
          return console.log(res.msg); // temp Let's check the error !
        })
        .catch(err => console.log(err)) // temp Let's check the error
  };

  render(){
    const { uploaded, loading, videoId } = this.state;
    const { history } = this.props;

    if(loading) return <LoadingSpinner text="Processing Your Video"/>;

    if(uploaded && !loading) return <UploadEdit videoId={videoId} redirect={() => history.push("/") }/>;
    return <UploadPicker onUploadSuccess={this.onUploadSuccess}/>
  }
}

export default requireLoggedIn(UploadContainer);
