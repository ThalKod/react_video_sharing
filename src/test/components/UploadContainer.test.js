import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { UploadContainer } from "components/Upload/UploadContainer";

describe("LoadingSpinner", () => {
  it("should correctly render the UploadPicker components", () => {
    const wrapper = shallow(<UploadContainer />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find("UploadPicker").length).toBe(1);
  });

  it("should correctly render the Loading Spinner", () => {
    const wrapper = shallow(<UploadContainer />);

    wrapper.setState({ loading: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should correctly render the UploadEdit components", () => {
    const wrapper = shallow(<UploadContainer />);

    wrapper.setState({ uploaded: true });
    expect(wrapper.find("UploadEdit").length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
