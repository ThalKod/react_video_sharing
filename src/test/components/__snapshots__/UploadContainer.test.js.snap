// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner should correctly render the Loading Spinner 1`] = `
<_default
  text="Processing Your Video"
/>
`;

exports[`LoadingSpinner should correctly render the UploadEdit components 1`] = `
<UploadEdit
  redirect={[Function]}
  videoId={null}
/>
`;

exports[`LoadingSpinner should correctly render the UploadPicker components 1`] = `
<UploadPicker
  onUploadSuccess={[Function]}
/>
`;
